src/ml/neural_net/tf_compute_context.cpp

```diff
@@ -155,7 +155,16 @@ float_array_map tf_model_backend::export_weights() const {
   float_array_map result;
   call_pybind_function([&]() {
     // Call export_weights method on the TensorFLowModel
-    pybind11::object exported_weights = model_.attr("export_weights")();
+    pybind11::dict exported_weights = model_.attr("export_weights")();
+
+    // it should be trivial to call this if we use the same interpreter process
+    pybind11::module np = pybind11::module::import("numpy");
+    for (auto& kv : exported_weights) {
+      // defensively call ascontiguousarray to force to reorganize
+      // underlying numpy memory layout using the real strides
+      exported_weights[kv.first] = np.attr("ascontiguousarray")(kv.second);
```

Comment:
If we are doing it here, I think we should remove it from export weights from AC/OD and any other toolkits that have it in `export_weights` in `tf_model_architecture` file.

Reply:
Could you explain what we should remove?

Reply:
So I have had the same problem earlier. I was tackling it by calling it here https://github.com/apple/turicreate/blob/aedc7a9868ffe9df4509cd0c5adbb7949774f8bd/src/python/turicreate/toolkits/activity_classifier/_tf_model_architecture.py#L289
Similarly, for OD also I was calling it in the python file. Your fix is better as it will do it for all toolkits together. So can you also remove these files along with this change since this change is better than what I was doing?

Reply:
Ha, you don't have to worry about it. IMO, your change is good (be able to track down this issue is really excellent). If I had known you met this issue before, I would ask you and add `ascontiguousarray` into python file without any hesitation.

Back to the point, this is a defensive programming paradigm just to protect us from people forget to call `ascontiguousarray` accidentally. But this doesn't mean we don't need to call `ascontiguousarray` explicitly.

We should advocate adding `ascontiguousarray` eagerly in our python implementation to explicitly document to other contributors what we are doing here. It's better to let other contributors also not to worry about keeping `ascontiguousarray` in mind when exporting. Even though they forget to have it called, they won't jump into the quagmire of debugging the mismatched weights as we did before.

Reply:
Okay. I think you ran into it when I was away. Maybe should have documented it better. Good job tracking down the issue too.

Reply:
yeap! Nice to have you around ;-)

Reply:
Could this line be simplified to:
```
kv.second = np.attr("ascontiguousarray")(kv.second);
```
?

Reply:
nope. kv is a type of `pair<py::handle, py::handle>`. The py::handle is a pointer to an object without any ownership. We need py::object here so that why I'm using `operator[](py::handle)`.

reference:
[handle](https://pybind11.readthedocs.io/en/stable/reference.html#_CPPv46handle)
[index operator]https://pybind11.readthedocs.io/en/stable/reference.html#_CPPv4NK10object_apiixEPKc

Reply:
Ah, I see, we need to assign into the result of `operator[]` to trigger the call to Python-side `__setitem__`. Thanks for the link!